src/main/java/org/kitteh/irc/client/library/implementation/EventListener.java

```diff
@@ -496,6 +487,43 @@ public void cap(ClientReceiveCommandEvent event) {
         }
     }
 
+    @CommandFilter("CHGHOST")
+    @Handler(filters = @Filter(CommandFilter.Filter.class), priority = Integer.MAX_VALUE - 1)
+    public void chghost(ClientReceiveCommandEvent event) {
+        if (event.getParameters().size() != 2) {
+            throw new KittehServerMessageException(event.getOriginalMessage(), "Invalid number of parameters for CHGHOST message");
+        }
+
+        if (!(event.getActor() instanceof User)) {
+            throw new KittehServerMessageException(event.getOriginalMessage(), "Invalid actor for CHGHOST message");
+        }
+
+        User user = (User) event.getActor();
+        ActorProvider.IRCUser oldIrcUser = this.client.getActorProvider().getUser(user.getNick());
```

Comment:
Naming it `oldIrcUser` is confusing in that its really just reflecting the current user regardless of new/oldness. Maybe just call it `ircUser`?